include/evaluators/blas_functor_traits.hpp

```diff
@@ -0,0 +1,153 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) 2016 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename blas_functor_traits.hpp
+ *
+ **************************************************************************/
+
+#ifndef BLAS_FUNCTOR_TRAITS_HPP
+#define BLAS_FUNCTOR_TRAITS_HPP
+
+#include <iostream>
+#include <stdexcept>
+#include <vector>
+
+#include <CL/sycl.hpp>
+
+#include <executors/blas_packet_traits_sycl.hpp>
+#include <operations/blas_operators.hpp>
+
+namespace blas {
+
+template <class Functor, typename ScalarT, typename Device>
+struct functor_traits : Functor {
+  /* static constexpr bool supported = false; */
+};
+
+template <typename ScalarT, typename Device>
+struct functor_traits<iniAddOp1_struct, ScalarT, Device> : iniAddOp1_struct {
+  /* static constexpr bool supported = true; */
+};
```

Comment:
Should these functors be initialised somewhere? 
For instance. the initial value of the addition is zero